pkg/operators/filter/filter_test.go

```diff
@@ -0,0 +1,479 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filter
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	gadgetcontext "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/simple"
+)
+
+func TestFilterRuleExtractor(t *testing.T) {
```

Comment:
Perhaps we can test more error cases:

```bash 
		{
			name:      "dsname",
			filter:    "dsname:abc==def",
			dsName:    "dsname",
			fieldName: "abc",
			op:        comparisonTypeMatch,
			negate:    false,
			value:     "def",
		},
		{
			name:   "bad1",
			filter: "abc",
			error:  true,
		},
		{
			name:   "bad2",
			filter: "abc==",
			error:  true,
		},
		{
			name:   "bad3",
			filter: "abc===def",
			error:  true,
		},
		{
			name:   "bad4",
			filter: "abc!==def",
			error:  true,
		},
		{
			name:   "bad5",
			filter: "abc!def",
			error:  true,
		},
		{
			name:   "bad6",
			filter: "abc!",
			error:  true,
		},
		{
			name:   "bad7",
			filter: "abc:",
			error:  true,
		},
		{
			name:   "bad8",
			filter: ":",
			error:  true,
		},
```

(I'm wondering if `name` can be `filter` to avoid having to think about it)

Reply:
Thanks! - and yeah, makes sense.